src/libraries/Common/src/Interop/Browser/Interop.HttpHandlerService.cs

```diff
@@ -0,0 +1,535 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+using JSObject = Interop.JavaScript.JSObject;
+using JSException = Interop.JavaScript.JSException;
+using HostObject = Interop.JavaScript.HostObject;
+using Uint8Array = Interop.JavaScript.Uint8Array;
+using Function = Interop.JavaScript.Function;
+
+internal static partial class Interop
+{
+    internal static partial class Browser
+    {
+        internal interface IHttpHandlerService
+        {
+            Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken);
+            void Dispose();
+        }
+
+        public class BrowserHttpHandlerService : IHttpHandlerService, IDisposable
```

Comment:
Why is it in Common and not System.Net.Http?

Reply:
This whole implementation looks like it should just be in the body of BrowserHttpHandler.  The Interop top-level type should be reserved just for the interop parts, e.g. P/Invokes or the JavaScript equivalent, etc.

Reply:
This has been moved.  It was just easier to get this code moved over with it separate because it is a separate library in mono for wasm.